oak_functions/lookup/src/lib.rs

```diff
@@ -181,33 +174,52 @@ where
         test_manager
     }
 
-    pub fn update_extend(&self, new_data: Data) -> UpdateStatus {
+    pub fn extend_next_lookup_data(&self, new_data: Data) {
+        info!("Start extending next lookup data");
         let mut data_builder = self.data_builder.lock();
         data_builder.extend(new_data);
-        UpdateStatus::Extended
+        info!("Finish extending next lookup data");
     }
 
-    // Finishing the update from every state.
-    pub fn update_finish(&self, new_data: Data) -> UpdateStatus {
-        let mut data_builder = self.data_builder.lock();
-        data_builder.extend(new_data);
-        let next_data = data_builder.build();
-        let mut data = self.data.lock();
-        *data = Arc::new(next_data);
-        UpdateStatus::Finished
+    // Finish building the next lookup data and replace the current lookup data in place.
+    pub fn finish_next_lookup_data(&self) {
+        let data_len;
+        let next_data_len;
+        info!("Start replacing lookup data by next lookup data");
+        {
+            let mut data_builder = self.data_builder.lock();
+            let next_data = data_builder.build();
+            next_data_len = next_data.len();
+            let mut data = self.data.lock();
+            *data = Arc::new(next_data);
+            data_len = data.len();
+        }
+        info!(
+            "Finished replacing lookup data with len {} by next lookup data with len {}",
```

Comment:
@andrisaar Just to confirm: this is the kind of logging we are looking for?

Reply:
Yep, that should be useful!